mne/viz/evoked.py

```diff
@@ -446,18 +445,15 @@ def _handle_spatial_colors(locs3d, info, idx, ch_type, psd, ax):
         outlines = 'skirt'
     # drop channels that are not in the data
     used_nm = np.array(_clean_names(info['ch_names']))[idx]
-
     names = np.asarray([name for name in used_nm if name in layout.names])
-    name_idx = [layout.names.index(name) for name in names]
 
     # find indices for bads
     bads = [np.where(names == bad)[0][0] for bad in info['bads'] if bad in
             names]
-    pos, outlines = _check_outlines(layout.pos[:, :2], outlines, head_pos)
-    pos = pos[name_idx]
+    pos = _auto_topomap_coords(info, idx, to_sphere=True)
```

Comment:
@jona-sassenhagen you probably meant something like this. Now it looks like this 
![spatial](https://cloud.githubusercontent.com/assets/3456414/25661425/e0873b24-3010-11e7-92de-9e1966f46e42.png)